icu4c/source/i18n/gregoimp.cpp

```diff
@@ -57,11 +57,11 @@ double ClockMath::floorDivide(double numerator, int32_t denominator,
 }
 
 double ClockMath::floorDivide(double dividend, double divisor,
-                         double* remainder) {
+                         int32_t* remainder) {
     // Only designed to work for positive divisors
     U_ASSERT(divisor > 0);
     double quotient = floorDivide(dividend, divisor);
-    double r = dividend - (quotient * divisor);
+    int32_t r = dividend - (quotient * divisor);
```

Comment:
Is r guaranteed to fit into an int32_t?
Are we losing fraction digits? If so, is that harmless/desirable?